File: schrottverwaltung/src/main/java/de/fhdortmund/schrottverwaltung/eigentuemer/Eigentuemer.java
package de.fhdortmund.schrottverwaltung.eigentuemer;

import de.fhdortmund.schrottverwaltung.immobilie.AdresseT;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "eigentuemer")
@Getter
@Setter
public class Eigentuemer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "eigentuemer_id")
    private Long id;

    @Column(name = "vorname")
    private String vorname;

    @Column(name = "nachname")
    private String nachname;

    @ManyToOne
    @JoinColumn(name = "adresse_id")
    private AdresseT anschrift;

    private String anmerkung;
}

File: schrottverwaltung/src/main/kotlin/de/fhdortmund/schrottverwaltung/eigentuemer/Eigentuemer.kt
package de.fhdortmund.schrottverwaltung.eigentuemer

import de.fhdortmund.schrottverwaltung.immoblilie.AdresseT
import jakarta.persistence.*

@Entity
@Table(name = "eigentuemer")
data class Eigentuemer(
    @Column(name = "vorname")
    var vorname: String,
    @Column(name = "nachname")
    var nachname: String,
    @ManyToOne
    @JoinColumn(name = "adresse_id")
    var anschrift: AdresseT,
    var anmerkung: String,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "eigentuemer_id")
    var id: Long? = null
)
